lib/NetPacksLib.cpp

```diff
@@ -106,10 +106,10 @@ DLL_LINKAGE void SetCommanderProperty::applyGs(CGameState *gs)
 	switch (which)
 	{
 		case BONUS:
-			commander->accumulateBonus (accumulatedBonus);
+			commander->accumulateBonus (std::make_shared<Bonus>(accumulatedBonus));

```

Comment:
You should change accumulatedBonus declaration instead. NetPacks changes are freely allowed (for now).